scripts/e2e/android-devices.js

```diff
@@ -0,0 +1,110 @@
+const { getCommandLineResponse } = require('./command-line-helpers');
+
+const DEFAULT_CI_AVD_NAME = 'e2e_emulator';
+const isRunningCI = !!process.env.CI;
+
+const envVarKeys = /** @type {const} */ ({
+  adbSerial: 'RNS_DEVICE_SERIAL',
+  avdName: 'RNS_AVD_NAME',
+});
+
+const passedAdbSerial = process.env[envVarKeys.adbSerial];
+
+function resolveAttachedAndroidDeviceSerial() {
+  if (passedAdbSerial) return passedAdbSerial;
+  const connectedNonEmulators = getDeviceIds().filter(
```

Comment:
`connectedNonEmulators` -> `connectedPhysicalDevices` - no necessary to have a negation in the variable name